### ISSUE TITLE
[1.19.0] can't generate docs for angular2 anymore - analyzer NoSuchMethodError

### ISSUE THREAD

**Issue by chalin** [dart-lang/dartdoc#1233]
Having upgraded to 1.19.0, I can no longer build the API docs for angular2. When I try I get:
```
Generating documentation for 'angular2' into /Users/chalin/tmp/api/

Generation failed: Class 'JavaFile' has no instance getter 'path'.

NoSuchMethodError: method not found: 'path'
Receiver: Instance of 'JavaFile'
Arguments: []
dart:core                                                                                  Object.noSuchMethod
package:analyzer/source/pub_package_map_provider.dart 173                                  PubPackageMapProvider._runPubListDefault
package:analyzer/source/pub_package_map_provider.dart 74                                   PubPackageMapProvider.computePackageMap
package:dartdoc/dartdoc.dart 150                                                           DartDoc._parseLibraries
package:dartdoc/dartdoc.dart 93                                                            DartDoc.generateDocs.<async>
===== asynchronous gap ===========================
dart:async                                                                                 _Completer.completeError
package:dartdoc/dartdoc.dart 139                                                           DartDoc.generateDocs.<async>
===== asynchronous gap ===========================
dart:async                                                                                 Future.Future.microtask
package:dartdoc/dartdoc.dart                                                               DartDoc.generateDocs
/b/build/slave/dart-sdk-mac-stable/build/sdk/third_party/pkg/dartdoc/bin/dartdoc.dart 142  main.<async>.<fn>.<async>
===== asynchronous gap ===========================
package:stack_trace                                                                        Chain.capture
/b/build/slave/dart-sdk-mac-stable/build/sdk/third_party/pkg/dartdoc/bin/dartdoc.dart 141  main.<async>
dart:async                                                                                 _SyncCompleter.complete
package:dartdoc/dartdoc.dart 50                                                            initGenerators.<async>
dart:async                                                                                 _SyncCompleter.complete
package:dartdoc/src/html/html_generator.dart 67                                            HtmlGenerator.create.<async>
dart:async                                                                                 _SyncCompleter.complete
package:dartdoc/src/html/templates.dart 115                                                Templates.create.<async>
dart:async                                                                                 _SyncCompleter.complete
package:dartdoc/src/html/templates.dart 98                                                 Templates.create.<async>._loadTemplate.<async>
dart:async                                                                                 _SyncCompleter.complete
package:dartdoc/src/html/resource_loader.dart 22                                           loadAsString.<async>
dart:async                                                                                 _SyncCompleter.complete
package:dartdoc/src/html/resource_loader.dart 33                                           loadAsBytes.<async>
```

cc @kwalrath

**Comment by keertip** [comment]
Most likely caused by a change in analyzer api. Will take a look.

**Comment by keertip** [comment]
Talked to the analyzer folks, they are in the process of simplifying the creation of the analysis context and adding the necessary uri resolvers. This should be out in a day or two. So waiting on that.

**Comment by astashov** [comment]
Breaks all the the new docs on dartdocs.org as well:

https://www.dartdocs.org/history/index.html
https://www.dartdocs.org/documentation/stagehand/1.0.12/log.txt

**Comment by devoncarew** [comment]
This is fixed in master; we're waiting on a merge into the current stable SDK.
